app/src/test/java/com/antosik/benchproject/app/popular/movies/viewModel/PopularMoviesViewModelTest.kt

```diff
@@ -28,54 +30,43 @@ internal class PopularMoviesViewModelTest {
         Movie(10003, "Test3", 3.4, "2022-06-21", "https://image.tmdb.org/t/p/w500/something3.com")
     )
+    val movieId = 10000
     val getPopularMoviesUseCase: GetPopularMoviesUseCase = mockk()
-    val popularMoviesFragmentNavigator: PopularMoviesFragmentNavigator = mockk()
-
-    @Test
-    fun `when getPopularMoviesUseCase called and load data should set popularMovies`() = runTest {
-        coEvery { getPopularMoviesUseCase() } returns moviesModel
-        val tested = PopularMoviesViewModel(getPopularMoviesUseCase, popularMoviesFragmentNavigator)
-
-        tested.popularMovies.value shouldBeEqualTo movies
+    val popularMoviesFragmentNavigator: PopularMoviesFragmentNavigator = mockk {
+        every { navigateToMovieDetailsFragment(movieId) } just Runs
     }
-
-    @Test
-    fun `when getPopularMoviesUseCase called and load data, isLoaderVisible should be false`() = runTest {
-        coEvery { getPopularMoviesUseCase() } returns moviesModel
-        val tested = PopularMoviesViewModel(getPopularMoviesUseCase, popularMoviesFragmentNavigator)
-
-        tested.isLoaderVisible.value shouldBe false
+    val successState = UIState.Success(movies)
+    val loadingState = UIState.Loading
+    val emptyState = UIState.Empty
+    val tested by lazy {
+        PopularMoviesViewModel(getPopularMoviesUseCase, popularMoviesFragmentNavigator)
     }
 
     @Test
-    fun `when getPopularMoviesUseCase called and load data, isPlaceholderVisible should be false`() = runTest {
+    fun `when getPopularMoviesUseCase called and load data, UIState should be in Success state`() {
         coEvery { getPopularMoviesUseCase() } returns moviesModel
-        val tested = PopularMoviesViewModel(getPopularMoviesUseCase, popularMoviesFragmentNavigator)
 
-        tested.isPlaceholderVisible.value shouldBe false
+        tested.popularMoviesViewState.value shouldBeEqualTo successState
     }
 
     @Test
-    fun `when getPopularMoviesUseCase called and return empty list, isPlaceholderVisible should be true`() = runTest {
+    fun `when getPopularMoviesUseCase called and return empty list, UIState should be in Empty state`() {
         coEvery { getPopularMoviesUseCase() } returns emptyList()
-        val tested = PopularMoviesViewModel(getPopularMoviesUseCase, popularMoviesFragmentNavigator)
 
-        tested.isPlaceholderVisible.value shouldBe true
+        tested.popularMoviesViewState.value shouldBeEqualTo emptyState
     }
 
     @Test
-    fun `when getPopularMoviesUseCase called and load data, isRefreshing should be false`() = runTest {
-        coEvery { getPopularMoviesUseCase() } returns moviesModel
-        val tested = PopularMoviesViewModel(getPopularMoviesUseCase, popularMoviesFragmentNavigator)
+    fun `when getPopularMoviesUseCase called and working, UIState should be in Loading state`() {
+        coEvery { getPopularMoviesUseCase() } just Awaits
```

Comment:
Ideally would be to use the benefits of dynamic tests here, maybe there is a way to handle this somehow

Reply:
IMO done 😅